fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/DefaultItemComponentEvents.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.item.v1;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import net.minecraft.component.ComponentMap;
+import net.minecraft.component.DataComponentType;
+import net.minecraft.item.Item;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Events to modify the default {@link ComponentMap} of items.
+ */
+public final class DefaultItemComponentEvents {
+	/**
+	 * Event used to add or remove data components to known items.
+	 */
+	public static final Event<ModifyCallback> MODIFY = EventFactory.createArrayBacked(ModifyCallback.class, listeners -> context -> {
+		for (ModifyCallback listener : listeners) {
+			listener.modify(context);
+		}
+	});
+
+	/**
+	 * Event used to modify the default data components of items after they have been modified by other mods during {@link #MODIFY}.
+	 */
+	public static final Event<AfterModifyCallback> AFTER_MODIFY = EventFactory.createArrayBacked(AfterModifyCallback.class, listeners -> context -> {
```

Comment:
Why two events?

Reply:
The idea was to allow mods to add based on other mods additions, I think I can move it into one but with some pref overhead. As it only happens once it should be fine. It's on my todo list to look at, but 1.21 comes first

Reply:
From my understanding, the first is for modifying default components of specific (known) items or groups of items, whereas the second one is for modifying every default component of a certain type. Doing so before all specific items have been modified is undesirable in the great majority of cases, so splitting it into two dedicated phases makes sense. I was on the fence about whether to suggest having the javadoc clarify this in greater detail.

Reply:
I think it can be done as one if each call to modify iterates all the items, then you can just use event priorities to ensure you go later.

Reply:
Now 1 event, makes the code much simpler even if it is theoretically slower.